src/index/coinstatsindex.cpp

```diff
@@ -0,0 +1,364 @@
+        if (!UndoReadFromDisk(block_undo, pindex)) {
+            return false;
+        }
+
+        std::pair<uint256, DBVal> read_out;
+        if (!m_db->Read(DBHeightKey(pindex->nHeight - 1), read_out)) {
+            return false;
+        }
+
+        uint256 expected_block_hash = pindex->pprev->GetBlockHash();
+        if (read_out.first != expected_block_hash) {
+            return error("%s: previous block header belongs to unexpected block %s; expected %s",
+                         __func__, read_out.first.ToString(), expected_block_hash.ToString());
+        }
+
+        bool is_bip30_block = (pindex->nHeight==91722 && pindex->GetBlockHash() == uint256S("0x00000000000271a2dc26e7667f8419f2e15416dc6955e5a6c6cdf3f2574dd08e")) ||
+                              (pindex->nHeight==91812 && pindex->GetBlockHash() == uint256S("0x00000000000af0aed4792b1acee3d966af36cf5def14935db8de83d6f9306f2f"));
+
+        // Add the new utxos created from the block
+        for (size_t i = 0; i < block.vtx.size(); ++i) {
+            const auto& tx = block.vtx.at(i);
+
+            // Skip duplicate txid coinbase transactions (BIP30).
+            if (is_bip30_block && tx->IsCoinBase()) {
+                m_unspendable_amount += block_subsidy;
+                continue;
+            }
+
+            for (size_t j = 0; j < tx->vout.size(); ++j) {
+                const CTxOut& out = tx->vout[j];
+                Coin coin = Coin(out, pindex->nHeight, tx->IsCoinBase());
+
+                total_out += coin.out.nValue;
+
+                // Skip unspendable coins
+                if (coin.out.scriptPubKey.IsUnspendable()) {
+                    m_unspendable_amount += coin.out.nValue;
+                    continue;
+                }
+
+                m_transaction_output_count++;
+                m_total_amount += coin.out.nValue;
+                m_bogo_size += GetBogoSize(coin.out.scriptPubKey);
+            }
+
+            // The coinbase tx has no undo data since no former output is spent
+            if (i > 0) {
+                const auto& tx_undo = block_undo.vtxundo.at(i-1);
+
+                for (size_t j = 0; j < tx_undo.vprevout.size(); ++j) {
+                    Coin coin = tx_undo.vprevout[j];
+
+                    total_in += coin.out.nValue;
+
+                    m_transaction_output_count--;
+                    m_total_amount -= coin.out.nValue;
+                    m_bogo_size -= GetBogoSize(coin.out.scriptPubKey);
+                }
+            }
+        }
+    } else {
+        // Genesis block
+        m_unspendable_amount += block_subsidy;
+    }
+
+    m_coinbase_amount = block_subsidy;
+    m_total_prevout_spent_amount = total_in;
+
+    if (pindex->nHeight > 0) {
+        m_total_new_outputs_ex_coinbase_amount = total_out - block_subsidy;
+
+        // Unclaimed block rewards
+        if ((total_in + block_subsidy) > total_out) {
+            m_unspendable_amount += (total_in + block_subsidy - total_out);
```

Comment:
I think this is causing BIP30 blocks to be double counted as unspendable

`bitcoin-cli gettxoutsetinfo 'none' 91722 true`

`"unspendable_amount": 100.00000000`